### PR TITLE
Datadog tracer

### PR DESCRIPTION
**Pull request by eikmadsen**
Add an optional datadog tracer.

### CODE REVIEW

File: conf/conf.stub.toml [hunk 1]
```diff
@@ -157,6 +157,7 @@ Enabled = true
 OpTimeout = "1s"
 
 [NewRelic]
+Disabled = true
```
[the remark this answers is not in the document]
Comment (eikmadsen) [reply conf/conf.stub.toml:160]: Yes, the empty license disables it. This makes it easier to switch it on and off

File: api/datadog.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+	})
+}
+
+// DatadogTx creates a request datadog middleware.
+func DatadogTx() goa.Middleware {
+	return func(h goa.Handler) goa.Handler {
+		if !ddApp.Enabled() {
+			return func(ctx context.Context, rw http.ResponseWriter, req *http.Request) error {
+				return h(ctx, rw, req)
+			}
```
[the remark this answers is not in the document]
Comment (eikmadsen) [reply api/datadog.go:69]: Hmm, that would call the handler.  Maybe I could just `return h` :thinking:

File: conf/conf.stub.toml [hunk 3]
```diff
@@ -170,3 +171,5 @@ ExcludeAttributes = ["password","header_Authorization","header_Cookie"]
 # Relic UI.  This is an optional setting.
 HostDisplayName = ""
 
+[Datadog]
+Enabled = true
```
[the remark this answers is not in the document]
Comment (eikmadsen) [reply conf/conf.stub.toml:175]: That is what I did initially. But didn't wan't to disable New Relic by default if someone depends on it.

File: api/datadog.go [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+				}
+			}
+			var ninfo int
+			var mu sync.Mutex
+			iinfo := func(msg string, keyvals ...interface{}) {
+				mu.Lock()
+				n := ninfo
+				ninfo++
+				mu.Unlock()
+				if n < 10 {
```
[the remark this answers is not in the document]
Comment (eikmadsen) [reply api/datadog.go:106]: We have the same limit in newrelic. Not sure it is worth effort to make this configuable.

File: api/datadog.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (klauspost) [reply api/datadog.go]: Good idea, if you move it outside 👍🏼

File: conf/conf.stub.toml (hunk 3 above)
[the remark this answers is not in the document]
Comment (eikmadsen) [reply conf/conf.stub.toml:175]: makes sense, I'll do an update to fix this :-)

**Comment by kirkeby** [comment]
LGTM :+1:

**Review by klauspost — APPROVED** [review]
LGTM

**Pull request merged**